main.py

```diff
@@ -206,7 +286,7 @@ def genArg(settings, name, port):
 		return args
 
 	@staticmethod
-	def generateLaunchCmd(settings, skipPathChecks=False):
+	def generateLaunchCmd(settings, main, skipPathChecks=False):
```

Comment:
This is ugly. Why does it need full access to the `main` object? Isn't it enough to pass the path to the DVD image?

Generally, I believe we'll probably have some "virtual-xbox" state object, which basically says: "boot a virtual-xbox, which was left in this condition: revision 1.0, 128MB, bios flashed with XYZ, Halo disc in DVD tray, ..." (so a combination of settings and - possibly - user input).